src/lib/rengo_utils.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2012-2022  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import swal from "sweetalert2";
+import { put, post, del } from "requests";
+import { errorAlerter } from "misc";
+import { _ } from "translate";
+
+type Challenge = socket_api.seekgraph_global.Challenge;
+type RengoParticipantsDTO = rest_api.RengoParticipantsDTO;
+
+// This is used by the SeekGraph to perform this function, as well as the Play page...
+export function nominateForRengoChallenge(c: Challenge): Promise<RengoParticipantsDTO> {
+    swal({
+        text: _("Joining..."), // translator: the server is processing their request to join a rengo game
+        type: "info",
+        showCancelButton: false,
+        showConfirmButton: false,
+        allowEscapeKey: false,
+    }).catch(swal.noop);
+
+    return put("challenges/%%/join", c.challenge_id, {})
+        .then((res) => {
+            swal.close();
+            return res;
+        })
+        .catch((err: any) => {
+            swal.close();
+            errorAlerter(err);
+        });
+}
+
+export function assignToTeam(
+    player_id: number,
+    team: string,
+    challenge,
+): Promise<RengoParticipantsDTO> {
+    const assignment =
+        team === "rengo_black_team"
+            ? "assign_black"
+            : team === "rengo_white_team"
+            ? "assign_white"
+            : "unassign";
+
+    return put("challenges/%%/team", challenge.challenge_id, {
+        [assignment]: [player_id], // back end expects an array of changes, but we only ever send one at a time.
+    })
+        .then((res) => {
+            return res;
+        })
+        .catch((err) => {
```

Comment:
Nice - I will tweak things this way.



> ESLint won't complain because the promise is getting returned, and the callers are forced to do the catch anyway.

Do you know what the deal with onClick handlers is?    It seems that if you pass them a function returning a promise, ES lint thinks everything is OK.

Reply:
Oh I see. I didn't realize rengo_utils.assignToTeam gets passed to onClick directly in one case. I only saw where it gets called and re-caught in InviteList. So maybe never mind on the catch suggestion.

>Do you know what the deal with onClick handlers is? It seems that if you pass them a function returning a promise, ES lint thinks everything is OK.

`no-floating-promises` only warns when the promise is floating.  Not sure there's an ESLint rule to do what you're saying (although i do see how it would be useful)